tensorpipe/common/cuda.h

```diff
@@ -125,14 +127,67 @@ inline int cudaDeviceForPointer(const void* ptr) {
   return attrs.device;
 }
 
-using CudaPinnedBuffer = std::shared_ptr<uint8_t>;
+class CudaPinnedBufferAllocator {
+ public:
+  CudaPinnedBufferAllocator(const CudaPinnedBufferAllocator&) = delete;
+  CudaPinnedBufferAllocator(CudaPinnedBufferAllocator&&) = delete;
+  CudaPinnedBufferAllocator& operator=(const CudaPinnedBufferAllocator&) =
+      delete;
+  CudaPinnedBufferAllocator& operator=(CudaPinnedBufferAllocator&&) = delete;
+
+  CudaPinnedBufferAllocator(size_t numChunks = 1024, size_t chunkSize = 1024)
```

Comment:
These defaults are arbitrary for now. We should figure out sensible values, and expose those parameters as options to the context.